# Pós-graduação em Ciência de Dados e Analytics
### MVP: Disciplina Sprint Engenharia de Dados

####Atividade: Construção de pipeline para análise de dados e resolução de problema.####

**Aluna:** Ana Nery Alves Daniel <br><br>

#####Problema pressuposto:#####

Baixo quantitativo de sedes administrativas de instituições de ensino superior (IES) da rede pública no estado do Ceará, bem como aumentar o número de docentes com doutorado nessas instituições.

**Objetivo:**

<p align="justify">Realizar levantamento de quais municípios do estado do Ceará possuem sede administrativa de instituições de ensino superior (IES), bem como identificar o número de docentes em exercício com nível de doutorado nessas sedes, identificando, ainda, o quantitativo de docentes declarados pretos e índios, com vistas a expandir políticas afirmativas para esses grupos e ampliar o número de IES no estado.</p>

> 
Ao final do levantamento, será sugerida a ampliação de sede de instituições de ensino superior por Região de Planejamento do estado do Ceará*.
> 
**Observação:** O foco deste levantamento não inclui unidades menores das instituições de ensino superior (polos). Portanto, considera-se IES neste levantamento somente as sedes administrativas das instituições de ensino superior no estado do Ceará. 


<span style="font-size:5px;">*Divisões territoriais criadas para organizar o planejamento e a implementação de políticas públicas. Classificação do Instituto de Pesquisa e Estratégia Econômica do Ceará - IPECE.</span>




#### Ingestão dos Dados
- [Copiar dataset dos microdados da Educação Superior 2023](https://www.gov.br/inep/pt-br/acesso-a-informacao/dados-abertos/microdados/censo-da-educacao-superior)
- [Copiar dataset dos Municípios do Ceará segundo Regiões de Planejamento](https://www.ipece.ce.gov.br/regioes-de-planejamento/)
- Enviar arquivos para o volume (workspace/default)
- Gerar tabelas a partir dos arquivos com dados brutos (workspace/default) renomeando as tabelas para: "regioes_planejamento_ceara" e "microdados_ed_sup_ies_2023"

In [0]:
%sql
-- TESTE DE CARGAS DE MICRODADOS DE EDUCAÇÃO SUPERIOR (FASE BRONZE)
SELECT 
* 
FROM default.microdados_ed_sup_ies_2023;


NU_ANO_CENSO,NO_REGIAO_IES,CO_REGIAO_IES,NO_UF_IES,SG_UF_IES,CO_UF_IES,NO_MUNICIPIO_IES,CO_MUNICIPIO_IES,IN_CAPITAL_IES,NO_MESORREGIAO_IES,CO_MESORREGIAO_IES,NO_MICRORREGIAO_IES,CO_MICRORREGIAO_IES,TP_ORGANIZACAO_ACADEMICA,TP_REDE,TP_CATEGORIA_ADMINISTRATIVA,IN_COMUNITARIA,IN_CONFESSIONAL,NO_MANTENEDORA,CO_MANTENEDORA,CO_IES,NO_IES,SG_IES,DS_ENDERECO_IES,DS_NUMERO_ENDERECO_IES,DS_COMPLEMENTO_ENDERECO_IES,NO_BAIRRO_IES,NU_CEP_IES,QT_TEC_TOTAL,QT_TEC_FUNDAMENTAL_INCOMP_FEM,QT_TEC_FUNDAMENTAL_INCOMP_MASC,QT_TEC_FUNDAMENTAL_COMP_FEM,QT_TEC_FUNDAMENTAL_COMP_MASC,QT_TEC_MEDIO_FEM,QT_TEC_MEDIO_MASC,QT_TEC_SUPERIOR_FEM,QT_TEC_SUPERIOR_MASC,QT_TEC_ESPECIALIZACAO_FEM,QT_TEC_ESPECIALIZACAO_MASC,QT_TEC_MESTRADO_FEM,QT_TEC_MESTRADO_MASC,QT_TEC_DOUTORADO_FEM,QT_TEC_DOUTORADO_MASC,IN_ACESSO_PORTAL_CAPES,IN_ACESSO_OUTRAS_BASES,IN_ASSINA_OUTRA_BASE,IN_REPOSITORIO_INSTITUCIONAL,IN_BUSCA_INTEGRADA,IN_SERVICO_INTERNET,IN_PARTICIPA_REDE_SOCIAL,IN_CATALOGO_ONLINE,QT_PERIODICO_ELETRONICO,QT_LIVRO_ELETRONICO,QT_DOC_TOTAL,QT_DOC_EXE,QT_DOC_EX_FEMI,QT_DOC_EX_MASC,QT_DOC_EX_SEM_GRAD,QT_DOC_EX_GRAD,QT_DOC_EX_ESP,QT_DOC_EX_MEST,QT_DOC_EX_DOUT,QT_DOC_EX_INT,QT_DOC_EX_INT_DE,QT_DOC_EX_INT_SEM_DE,QT_DOC_EX_PARC,QT_DOC_EX_HOR,QT_DOC_EX_0_29,QT_DOC_EX_30_34,QT_DOC_EX_35_39,QT_DOC_EX_40_44,QT_DOC_EX_45_49,QT_DOC_EX_50_54,QT_DOC_EX_55_59,QT_DOC_EX_60_MAIS,QT_DOC_EX_BRANCA,QT_DOC_EX_PRETA,QT_DOC_EX_PARDA,QT_DOC_EX_AMARELA,QT_DOC_EX_INDIGENA,QT_DOC_EX_COR_ND,QT_DOC_EX_BRA,QT_DOC_EX_EST,QT_DOC_EX_COM_DEFICIENCIA
2023,Centro-Oeste,5,Mato Grosso,MT,51,Cuiabá,5103403,1,Centro-Sul Mato-grossense,4,Cuiabá,17,1,1,1,0,0,FUNDACAO UNIVERSIDADE FEDERAL DE MATO GROSSO,1,1,UNIVERSIDADE FEDERAL DE MATO GROSSO,UFMT,Avenida Fernando Correa da Costa,2367,Cidade Universitária,Boa Esperança,78060900,1335,2,15,1,11,27,59,87,106,331,327,172,107,55,35,1,0,1,1,1,1,1,1,50,50,1742,1588,764,824,0,83,51,191,1263,1537,1276,261,51,0,44,144,298,318,276,197,136,175,1003,85,454,38,6,2,1559,29,12
2023,Centro-Oeste,5,Distrito Federal,DF,53,Brasília,5300108,1,Distrito Federal,1,Brasília,1,1,1,1,0,0,FUNDACAO UNIVERSIDADE DE BRASILIA,2,2,UNIVERSIDADE DE BRASÍLIA,UNB,Campus Universitário Darcy Ribeiro,s/n,-,Asa Norte,70910900,2906,0,0,2,24,100,118,177,225,746,704,386,265,93,66,1,0,1,1,1,1,1,1,441,13095,3120,2801,1288,1513,0,9,18,224,2550,2647,2323,324,154,0,44,146,335,500,482,369,377,548,1958,109,625,66,11,32,2669,132,10
2023,Nordeste,2,Sergipe,SE,28,São Cristóvão,2806701,0,Leste Sergipano,3,Aracaju,11,1,1,1,0,0,FUNDACAO UNIVERSIDADE FEDERAL DE SERGIPE,3,3,UNIVERSIDADE FEDERAL DE SERGIPE,UFS,Avenida Marechal Rondon,s/n,-,Jardim Rosa Elze,49100000,1769,0,23,8,12,211,137,269,188,275,187,219,137,65,38,1,0,1,1,1,1,1,1,40939,320055,1820,1768,846,922,0,100,2,210,1456,1652,1441,211,115,1,36,127,258,359,331,250,202,205,242,36,295,11,1,1183,1739,29,5
2023,Norte,1,Amazonas,AM,13,Manaus,1302603,1,Centro Amazonense,3,Manaus,7,1,1,1,0,0,FUNDACAO UNIVERSIDADE DO AMAZONAS,4,4,UNIVERSIDADE FEDERAL DO AMAZONAS,UFAM,Av. Rodrigo Otávio,6200,Unidade Sede,Coroado II,69077000,1931,9,54,5,40,116,201,225,258,389,266,169,134,35,30,1,0,1,1,0,1,1,1,41,6272,2051,2048,934,1114,0,94,200,537,1217,1868,1466,402,180,0,112,217,328,344,295,248,249,255,167,22,1684,6,2,167,2048,0,6
2023,Nordeste,2,Piauí,PI,22,Teresina,2211001,1,Centro-Norte Piauiense,2,Teresina,3,1,1,1,0,0,FUNDACAO UNIVERSIDADE FEDERAL DO PIAUI,14054,5,UNIVERSIDADE FEDERAL DO PIAUÍ,UFPI,Campus Universitário Ministro Petrônio Portela,s/n,SG - 07,Ininga,64049550,2243,128,223,58,82,340,508,51,71,258,211,135,99,44,35,1,0,1,1,1,1,1,1,0,0,1693,1469,710,759,2,5,70,250,1142,1374,1174,200,95,0,22,134,231,304,228,144,163,243,164,41,257,3,1,1003,1469,0,0
2023,Sudeste,3,Minas Gerais,MG,31,Ouro Preto,3146107,0,Metropolitana de Belo Horizonte,7,Ouro Preto,33,1,1,1,0,0,UNIVERSIDADE FEDERAL DE OURO PRETO,6,6,UNIVERSIDADE FEDERAL DE OURO PRETO,UFOP,Rua Diogo de Vasconcelos,122,-,Centro,35400000,653,3,19,1,10,17,41,34,64,90,109,108,96,38,23,0,0,0,1,1,1,1,1,60,28957,910,861,353,508,0,0,16,

In [0]:
%sql
-- TESTE DE CARGA DE MICRODADOS DOS MUNICÍPIOS DO ESTADO DO CEARÁ  (FASE BRONZE)

SELECT * 
FROM default.regioes_planejamento_ceara;

Código do município (IBGE),Nome do município,Região de Planejamento
2300101,Abaiara,Cariri
2300150,Acarape,Maciço de Baturité
2300200,Acaraú,Litoral Norte
2300309,Acopiara,Centro Sul
2300408,Aiuaba,Sertão dos Inhamuns
2300507,Alcântaras,Sertão de Sobral
2300606,Altaneira,Cariri
2300705,Alto Santo,Vale do Jaguaribe
2300754,Amontada,Litoral Oeste / Vale do Curu
2300804,Antonina do Norte,Cariri


#### Modelagem dos Dados

**Tipo de Modelagem:** Estrela

**Tabela Fato das Instituições de Ensino Superior**

**Nome da Tabela:** default.fato_ies

**Dicionário dos Dados**

| nome da coluna | tipo | tamanho | descricao |
| -------- | ----- | ----- | ----------- |
| CO_IES | Num | 8 | Código único de identificação da IES |
| NO_IES | Char | 200 | Nome da IES
| SG_IES |Char | 20 | Sigla da IES
| SG_UF_IES |Char | 2 | Sigla da Unidade da Federação da sede administrativa ou reitoria da IES
| CO_MUNICIPIO | Num | 7 | Código do município da sede administrativa da IES
| NO_TIPO_REDE | Char | 7 | Rede de ensino


In [0]:
%sql

-- CRIAR TABELA FATO  (FASE PRATA)

CREATE OR REPLACE TABLE default.fato_ies AS 
SELECT
  CO_IES,
  NO_IES,
  SG_IES,
  SG_UF_IES,
  CO_MUNICIPIO_IES AS CO_MUNICIPIO, 
CASE
  WHEN TP_REDE = 1 THEN 'Pública'
  WHEN TP_REDE = 2 THEN 'Privada'
END AS NO_TIPO_REDE
FROM default.microdados_ed_sup_ies_2023;

SELECT
  count(*)
FROM default.fato_ies;

count(1)
2580


**Tabela Dimensão das Regiões de Planejamento do Ceará**

**Nome da Tabela:** default.dim_regiao_planejamento_ce

**Dicionário de Dados**

| nome da coluna | tipo | tamanho | descricao |
| -------- | ----- | ----- | ----------- |
| CO_MUNICIPIO | Num| 7 | Código do município da sede administrativa da IES |
| NO_MUNICIPIO | Char | 150 | Nome do Município da sede administrativa ou reitoria da IES
| REG_PLA_CE | Char | 100 | Nome da região de planejamento do Ceará.




In [0]:
%sql

-- CRIAR TABELA DIMENSÃO REGIÕES DE PLANEJAMENTO DO CEARÁ (FASE PRATA)

CREATE OR REPLACE TABLE default.dim_regiao_planejamento_ce AS 
SELECT
  reg.`Código do município (IBGE)` AS CO_MUNICIPIO,
  reg.`Nome do município` AS NO_MUNICIPIO,
  reg.`Região de Planejamento` AS REG_PLA_CE
FROM default.regioes_planejamento_ceara AS reg;

SELECT 
COUNT(*)
FROM default.dim_regiao_planejamento_ce;

count(1)
184


**Tabela Dimensão dos Docentes**

**Nome da Tabela:** default.dim_docentes

**Dicionário de Dados**

| nome da coluna | tipo | tamanho | descricao |
| -------- | ----- | ----- | ----------- |
| CO_IES | Num | 8 | Código único de identificação da IES |
| QT_DOC_TOTAL | Num | 8 | Quantidade total de docentes (em exercício e afastados)
| QT_DOC_EXE | Num | 8 | Quantidade total de docentes em exercício
| QT_DOC_EX_DOUT | Num | 8 | Quantidade de docentes em exercício com doutorado
| QT_DOC_EX_PRETA | Num | 8 | Quantidade de docentes em exercício - Cor/Raça preta
| QT_DOC_EX_INDIGENA | Num | 8 | Quantidade de docentes em exercício - Cor/Raça indígena


In [0]:
%sql

-- CRIAR TABELA DIMENSÃO DOS DOCENTES (FASE PRATA)

CREATE OR REPLACE TABLE default.dim_docentes AS 
SELECT 
    CO_IES,
    QT_DOC_TOTAL,
    QT_DOC_EXE,
    QT_DOC_EX_DOUT,
    QT_DOC_EX_PRETA,
    QT_DOC_EX_INDIGENA
FROM default.microdados_ed_sup_ies_2023;


SELECT
*
FROM default.dim_docentes;


CO_IES,QT_DOC_TOTAL,QT_DOC_EXE,QT_DOC_EX_DOUT,QT_DOC_EX_PRETA,QT_DOC_EX_INDIGENA
1,1742,1588,1263,85,6
2,3120,2801,2550,109,11
3,1820,1768,1456,36,1
4,2051,2048,1217,22,2
5,1693,1469,1142,41,1
6,910,861,769,26,3
7,1339,1339,1282,13,0
8,1302,1302,1153,24,5
9,1682,1651,1311,26,1
10,1308,1308,781,13,1


#### Análise dos Dados

**BLOCO 1: PANORAMA DE IES POR **MUNICÍPIO****

**1. Quais municípios do estado do Ceará têm IES?**

In [0]:
%sql

SELECT 
  DISTINCT reg.NO_MUNICIPIO
FROM default.fato_ies ies
LEFT JOIN default.dim_regiao_planejamento_ce reg
  ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
WHERE reg.CO_MUNICIPIO IS NOT NULL
ORDER BY reg.NO_MUNICIPIO ASC;

NO_MUNICIPIO
Alto Santo
Aracati
Baturité
Camocim
Canindé
Cascavel
Caucaia
Crateús
Crato
Croatá


**2. Quais municípios do estado do Ceará não têm IES?**

In [0]:
%sql

SELECT 
  DISTINCT reg.NO_MUNICIPIO
FROM default.dim_regiao_planejamento_ce reg
LEFT JOIN default.fato_ies ies
  ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
WHERE ies.CO_IES IS NULL
ORDER BY reg.NO_MUNICIPIO ASC;

NO_MUNICIPIO
Abaiara
Acarape
Acaraú
Acopiara
Aiuaba
Alcântaras
Altaneira
Amontada
Antonina do Norte
Apuiarés


**3. Quais municípios do estado do Ceará têm IES da rede pública?**

In [0]:
%sql

SELECT 
  DISTINCT reg.NO_MUNICIPIO
FROM default.fato_ies ies
RIGHT JOIN default.dim_regiao_planejamento_ce reg
  ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
WHERE ies.NO_TIPO_REDE = 'Pública'
ORDER BY reg.NO_MUNICIPIO ASC;


NO_MUNICIPIO
Crato
Fortaleza
Juazeiro do Norte
Redenção
Sobral



**4. Quais municípios do estado do Ceará tem IES da rede privada?**


In [0]:
%sql

SELECT 
  DISTINCT reg.NO_MUNICIPIO
FROM default.fato_ies ies
RIGHT JOIN default.dim_regiao_planejamento_ce reg
  ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
WHERE ies.NO_TIPO_REDE = 'Privada'
ORDER BY reg.NO_MUNICIPIO ASC;

NO_MUNICIPIO
Alto Santo
Aracati
Baturité
Camocim
Canindé
Cascavel
Caucaia
Crateús
Crato
Croatá


**BLOCO 2: PANORAMA DE IES POR REGIÃO DE PLANEJAMENTO DO ESTADO DO CEARÁ**

![Mapa de Regiões de Planejamento do estado do Ceará](http://www2.ipece.ce.gov.br/atlas/capitulo1/11/pdf/Regioes_Planejamento2025.pdf)



**5. Quais regiões de planejamento do Ceará têm IES?**

In [0]:
%sql

SELECT 
  DISTINCT reg.REG_PLA_CE
FROM default.fato_ies ies
RIGHT JOIN default.dim_regiao_planejamento_ce reg
  ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
ORDER BY reg.REG_PLA_CE ASC;


REG_PLA_CE
Cariri
Centro Sul
Grande Fortaleza
Litoral Leste
Litoral Norte
Litoral Oeste / Vale do Curu
Maciço de Baturité
Serra da Ibiapaba
Sertão Central
Sertão de Canindé


**6. Quais regiões de planejamento do Ceará não têm IES?**

In [0]:
%sql

SELECT 
  reg.REG_PLA_CE
FROM default.dim_regiao_planejamento_ce reg
RIGHT JOIN default.fato_ies ies
  ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
WHERE reg.CO_MUNICIPIO IS NOT NULL
GROUP BY reg.REG_PLA_CE
  HAVING count(reg.REG_PLA_CE) < 1
;


REG_PLA_CE


**7. Quais regiões de planejamento do Ceará têm IES da rede pública?**

In [0]:
%sql

SELECT 
  DISTINCT reg.REG_PLA_CE
FROM default.fato_ies ies
RIGHT JOIN default.dim_regiao_planejamento_ce reg
  ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
WHERE ies.NO_TIPO_REDE = 'Pública'
ORDER BY reg.REG_PLA_CE ASC;

REG_PLA_CE
Cariri
Grande Fortaleza
Maciço de Baturité
Sertão de Sobral


**8. Quais regiões de planejamento do Ceará não têm IES da rede pública?**

In [0]:
%sql
SELECT 
  DISTINCT reg.REG_PLA_CE
FROM default.fato_ies ies
RIGHT JOIN default.dim_regiao_planejamento_ce reg
  ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
WHERE reg.REG_PLA_CE NOT IN (
  SELECT 
    DISTINCT reg.REG_PLA_CE
  FROM default.fato_ies ies
  RIGHT JOIN default.dim_regiao_planejamento_ce reg
    ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
  WHERE ies.NO_TIPO_REDE = 'Pública'
)
ORDER BY reg.REG_PLA_CE ASC;


REG_PLA_CE
Centro Sul
Litoral Leste
Litoral Norte
Litoral Oeste / Vale do Curu
Serra da Ibiapaba
Sertão Central
Sertão de Canindé
Sertão dos Crateús
Sertão dos Inhamuns
Vale do Jaguaribe


**9. Quais regiões de planejamento do Ceará têm IES da rede privada?**

In [0]:
%sql

SELECT 
  DISTINCT reg.REG_PLA_CE
FROM default.fato_ies ies
RIGHT JOIN default.dim_regiao_planejamento_ce reg
  ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
WHERE ies.NO_TIPO_REDE = 'Privada'
ORDER BY reg.REG_PLA_CE ASC;

REG_PLA_CE
Cariri
Centro Sul
Grande Fortaleza
Litoral Leste
Litoral Norte
Litoral Oeste / Vale do Curu
Maciço de Baturité
Serra da Ibiapaba
Sertão Central
Sertão de Canindé


**BLOCO 3: PANORAMA DE DOCENTES - QUANTITATIVO GERAL E DOCENTES COM DOUTORADO**

**10. Total de docentes em exercício na rede pública do Ceará**

In [0]:
%sql
SELECT 
  SUM(CAST(doc.QT_DOC_EXE AS INT)) AS Total_QT_DOC_EXE
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
WHERE ies.NO_TIPO_REDE = 'Pública'
AND ies.SG_UF_IES = 'CE';

Total_QT_DOC_EXE
7075


**11. Quantos docentes em exercício com doutorado existem na rede pública do Ceará por IES?**

In [0]:
%sql
SELECT 
  ies.NO_IES,
  doc.QT_DOC_EX_DOUT
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
WHERE ies.NO_TIPO_REDE = 'Pública'
AND ies.SG_UF_IES = 'CE'
ORDER BY CAST(doc.QT_DOC_EX_DOUT AS INT) DESC;



NO_IES,QT_DOC_EX_DOUT
UNIVERSIDADE FEDERAL DO CEARÁ,2001
UNIVERSIDADE ESTADUAL DO CEARÁ,785
"INSTITUTO FEDERAL DE EDUCAÇÃO, CIÊNCIA E TECNOLOGIA DO CEARÁ",673
UNIVERSIDADE DA INTEGRAÇÃO INTERNACIONAL DA LUSOFONIA AFRO-BRASILEIRA,364
UNIVERSIDADE FEDERAL DO CARIRI,275
UNIVERSIDADE REGIONAL DO CARIRI,257
UNIVERSIDADE ESTADUAL VALE DO ACARAÚ,233


**12. Total de docentes em exercício com doutorado na rede pública do Ceará**

In [0]:
%sql
SELECT 
  SUM(CAST(doc.QT_DOC_EX_DOUT AS INT)) AS Total_QT_DOC_EX_DOUT
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
WHERE ies.NO_TIPO_REDE = 'Pública'
AND ies.SG_UF_IES = 'CE';

Total_QT_DOC_EX_DOUT
4588


**13. Quantos docentes em exercício com doutorado existem na rede privada do Ceará por IES?**

In [0]:
%sql
SELECT 
  ies.NO_IES,
  doc.QT_DOC_EX_DOUT
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
WHERE ies.NO_TIPO_REDE = 'Privada'
AND ies.SG_UF_IES = 'CE'
ORDER BY CAST(doc.QT_DOC_EX_DOUT AS INT) DESC;

NO_IES,QT_DOC_EX_DOUT
UNIVERSIDADE DE FORTALEZA,461
CENTRO UNIVERSITÁRIO CHRISTUS,297
CENTRO UNIVERSITÁRIO INTA,164
Centro Universitário Estácio do Ceará,135
CENTRO UNIVERSITÁRIO ATENEU,85
Centro Universitário Fametro,71
CENTRO UNIVERSITÁRIO MAURÍCIO DE NASSAU DE FORTALEZA,66
CENTRO UNIVERSITÁRIO DOUTOR LEÃO SAMPAIO,61
Centro Universitário Fanor Wyden,55
FACULDADE DE MEDICINA ESTÁCIO DE JUAZEIRO DO NORTE,48


**14. Total de docentes em exercício com doutorado na rede privada do Ceará**

In [0]:
%sql
SELECT 
  SUM(CAST(doc.QT_DOC_EX_DOUT AS INT)) AS Total_QT_DOC_EX_DOUT
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
WHERE ies.NO_TIPO_REDE = 'Privada'
AND ies.SG_UF_IES = 'CE';

Total_QT_DOC_EX_DOUT
2216


**15. Quantos docentes em exercício com doutorado existem na rede pública do Ceará por região de planejamento?**

In [0]:
%sql

SELECT 
  reg.REG_PLA_CE,
  sum(doc.QT_DOC_EX_DOUT) as QT_DOC_EX_DOUT
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
RIGHT JOIN default.dim_regiao_planejamento_ce reg
  ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
WHERE ies.NO_TIPO_REDE = 'Pública'
GROUP BY reg.REG_PLA_CE
ORDER BY QT_DOC_EX_DOUT DESC

REG_PLA_CE,QT_DOC_EX_DOUT
Grande Fortaleza,3459.0
Cariri,532.0
Maciço de Baturité,364.0
Sertão de Sobral,233.0


**BLOCO 4: PANORAMA DE DOCENTES DECLARADOS PRETOS**

**16. Quantos docentes em exercício declaros pretos existem na rede pública do Ceará por IES?** 

In [0]:
%sql

SELECT 
  ies.NO_IES,
  ies.SG_IES,
  doc.QT_DOC_EX_PRETA
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
WHERE ies.NO_TIPO_REDE = 'Pública'
AND ies.SG_UF_IES = 'CE'
ORDER BY doc.QT_DOC_EX_PRETA DESC;

NO_IES,SG_IES,QT_DOC_EX_PRETA
UNIVERSIDADE FEDERAL DO CEARÁ,UFC,76
UNIVERSIDADE DA INTEGRAÇÃO INTERNACIONAL DA LUSOFONIA AFRO-BRASILEIRA,UNILAB,72
"INSTITUTO FEDERAL DE EDUCAÇÃO, CIÊNCIA E TECNOLOGIA DO CEARÁ",IFCE,51
UNIVERSIDADE ESTADUAL VALE DO ACARAÚ,UVA,33
UNIVERSIDADE FEDERAL DO CARIRI,UFCA,17
UNIVERSIDADE ESTADUAL DO CEARÁ,UECE,16
UNIVERSIDADE REGIONAL DO CARIRI,URCA,10


**17. Total de docentes em exercício declarados pretos na rede pública do Ceará** 

In [0]:
%sql
SELECT 
  SUM(CAST(doc.QT_DOC_EX_PRETA AS INT)) AS Total_QT_DOC_EX_PRETA
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
WHERE ies.NO_TIPO_REDE = 'Pública'
AND ies.SG_UF_IES = 'CE';


Total_QT_DOC_EX_PRETA
275


**18. Quantos docentes em exercício declarados pretos existem na rede pública do Ceará por região de planejamento?**

In [0]:
%sql
SELECT 
  reg.REG_PLA_CE,
  sum(doc.QT_DOC_EX_PRETA) as QT_DOC_EX_PRETA
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
RIGHT JOIN default.dim_regiao_planejamento_ce reg
  ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
WHERE ies.NO_TIPO_REDE = 'Pública'
GROUP BY reg.REG_PLA_CE
ORDER BY QT_DOC_EX_PRETA DESC
;

REG_PLA_CE,QT_DOC_EX_PRETA
Grande Fortaleza,143.0
Maciço de Baturité,72.0
Sertão de Sobral,33.0
Cariri,27.0


**19. Quantos docentes em exercício declarados pretos existem na rede privada por IES?**


In [0]:
%sql
SELECT 
  ies.NO_IES,
  doc.QT_DOC_EX_PRETA
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
WHERE ies.NO_TIPO_REDE = 'Privada'
AND ies.SG_UF_IES = 'CE'
ORDER BY 
  CASE 
    WHEN CAST(doc.QT_DOC_EX_PRETA AS INT) = 0 THEN -1
    ELSE CAST(doc.QT_DOC_EX_PRETA AS INT)
  END DESC;

NO_IES,QT_DOC_EX_PRETA
CENTRO UNIVERSITÁRIO DOUTOR LEÃO SAMPAIO,14
UNIVERSIDADE DE FORTALEZA,13
Centro Universitário Estácio do Ceará,12
CENTRO UNIVERSITÁRIO INTA,10
CENTRO UNIVERSITÁRIO ATENEU,8
Centro Universitário Fanor Wyden,6
CENTRO UNIVERSITÁRIO VALE DO SALGADO,6
CENTRO UNIVERSITÁRIO PARAÍSO,5
FACULDADE LUCIANO FEIJÃO,5
Faculdade Via Sapiens,5


**20. Total de docentes em exercício declarados pretos na rede privada.
[](url)**

In [0]:
%sql
SELECT 
  SUM(CAST(doc.QT_DOC_EX_PRETA AS INT)) AS Total_QT_DOC_EX_PRETA
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
WHERE ies.NO_TIPO_REDE = 'Privada'
AND ies.SG_UF_IES = 'CE';

Total_QT_DOC_EX_PRETA
158


**BLOCO 5: PANORAMA DE DOCENTES DECLARADOS INDÍGENAS**


**21. Quantos docentes em exercício declarados indígenas existem na rede pública do Ceará por IES?** 

In [0]:
%sql

SELECT 
  ies.NO_IES,
  doc.QT_DOC_EX_INDIGENA
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
WHERE ies.NO_TIPO_REDE = 'Pública'
AND ies.SG_UF_IES = 'CE'
ORDER BY 
  CASE 
    WHEN CAST(doc.QT_DOC_EX_INDIGENA AS INT) = 0 THEN -1
    ELSE CAST(doc.QT_DOC_EX_INDIGENA AS INT)
  END DESC;

NO_IES,QT_DOC_EX_INDIGENA
UNIVERSIDADE FEDERAL DO CEARÁ,15
UNIVERSIDADE FEDERAL DO CARIRI,4
UNIVERSIDADE ESTADUAL VALE DO ACARAÚ,3
UNIVERSIDADE REGIONAL DO CARIRI,3
"INSTITUTO FEDERAL DE EDUCAÇÃO, CIÊNCIA E TECNOLOGIA DO CEARÁ",3
UNIVERSIDADE DA INTEGRAÇÃO INTERNACIONAL DA LUSOFONIA AFRO-BRASILEIRA,2
UNIVERSIDADE ESTADUAL DO CEARÁ,0


**22. Total de docentes em exercício declarados indígenas na rede pública do Ceará.** 

In [0]:
%sql
SELECT 
  SUM(CAST(doc.QT_DOC_EX_INDIGENA AS INT)) AS Total_QT_DOC_EX_INDIGENA
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
WHERE ies.NO_TIPO_REDE = 'Pública'
AND ies.SG_UF_IES = 'CE';


Total_QT_DOC_EX_INDIGENA
30


**23. Quantos docentes em exercício declarados indigenas existem nas IES públicas do Ceará por região de planejamento?**

In [0]:
%sql
SELECT 
  reg.REG_PLA_CE,
  sum(doc.QT_DOC_EX_INDIGENA) as QT_DOC_EX_INDIGENA
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
RIGHT JOIN default.dim_regiao_planejamento_ce reg
  ON reg.CO_MUNICIPIO = ies.CO_MUNICIPIO
WHERE ies.NO_TIPO_REDE = 'Pública'
GROUP BY reg.REG_PLA_CE
ORDER BY QT_DOC_EX_INDIGENA DESC
;

REG_PLA_CE,QT_DOC_EX_INDIGENA
Grande Fortaleza,18.0
Cariri,7.0
Sertão de Sobral,3.0
Maciço de Baturité,2.0


**24. Total de docentes em exercício declarados indígenas na rede privada do Ceará.**

In [0]:
%sql

SELECT 
  SUM(CAST(doc.QT_DOC_EX_INDIGENA AS INT)) AS Total_QT_DOC_EX_INDIGENA
FROM default.fato_ies ies
LEFT JOIN default.dim_docentes doc
  ON doc.CO_IES = ies.CO_IES
WHERE ies.NO_TIPO_REDE = 'Privada'
AND ies.SG_UF_IES = 'CE';

Total_QT_DOC_EX_INDIGENA
17



### **Solução do Problema/Conclusão**


Considerando o problema pressuposto deste trabalho, o baixo quantitativo de sedes administrativas de instituições de ensino superior (IES) da rede pública no Ceará e a necessidade de aumentar o número de docentes com doutorado nessas sedes, observando dados que justifiquem a ampliação de políticas afirmativas para pessoas pretas e indígenas, a análise dos dados obtidos por meio de consulta SQL a partir de três tabelas criadas na modelagem estrela (Tabela FATO: Tabela_FATO_IES, Tabela Dimensão: Tabela_REGIAO_PLANEJAMENTO e Tabela Dimensão: Tabela_DOCENTES) evidenciou informações relevantes para o tratamento desse problema.

**No panorama de IES por município e região de planejamento**, a análise de dados revelou que dos 184 municípios do Ceará, somente 5 possuem IES da rede pública: Crato, Fortaleza, Juazeiro do Norte, Redenção e Sobral.

Apesar de existirem IES em todas as 14 regiões de planejamento do estado do Ceará, apenas em cinco delas existem IES da rede pública: Cariri, Grande Fortaleza, Maciço de Baturité e Sertão de Sobral. Isso corrobora com a necessidade de ampliação de IES públicas no estado, focando especialmente nas regiões sem nenhuma IES, maiores e mais distantes daquelas onde já existem essas instituições, tais como: Sertão Central, Sertão de Canindé, Sertão dos Crateús e Vale do Jaguaribe.

**No que concerne aos docentes com doutorado**, o  levantamento apontou que dos 7.075 docentes em exercício na rede pública, 4.588 possuem doutorado, ou seja, o quadro de doutores não chega a 70%. No entanto, na rede privada, esse valor é bem menor, com somente 2.216 profissionais com esse nível de formação. Isso evidencia que o nível de conhecimento do corpo docente das IES de rede pública é maior em termos quantitativos em relação aos docentes das IES de rede privada.

A análise de dados aponta também que o maior número de docentes doutores no Ceará está na região da Grande Fortaleza e o menor no Sertão de Sobral, indicativo importante para saber onde está a maior necessidade de ampliação desse nível de professores por região de planejamento.

Por fim, **no que se refere ao número de docentes declarados pretos na rede pública**, os dados apontaram uma queda brusca em relação ao quantitativo total de docentes, caindo de 7.075 (total) para 275 classificados nesse grupo, ou seja, valor inferior a 4% de representatividade, estando a maior parte de docentes declarados pretos concentrada na região da Grande Fortaleza (143) e a menor na região do Cariri (27).

**A representividade de docente declarada indígena**, por sua vez, é ainda menor, somente 30 docentes indígenas em um universo 7.075 professores, com a maior parte na Grande Fortaleza (18) e a menor no Maçiço de Baturiré (2).

**Em suma**, a análise dos dados apurados explicitam a necessidade de expansão de IES para outras regiões do estado, como Sertão Central, Sertão de Canindé, Sertão dos Crateús e Vale do Jaguaribe; bem como a necessidade de ampliar o número de doutores na região do Sertão de Sobral. Os resultados apontaram, ainda, a demanda de políticas afirmativas mais efetivas para ampliar o quantitativo de  docentes declarados preto e indígenas, como, por exemplo, um percentual maior de bolsas destinadas para esses grupos e a ampliação de cotas em concurso público com vistas a reparar essa diferença, avaliando a ampliação e redistribuição de vagas considerando também o quantitativo existente nas regiões de planejamentos do estado. 









### **Limitações**

O censo disponibilizado pelo Instituto Nacional de Estudos e Pesquisas Educacionais Anísio Teixeira (Inep) trazia somente dados referente às institutos de ensino superior como sede administrativa, o que impossibilitou que também fossem considerados neste levantamento os polos universitários do Ceará (unidades menores e com menos cursos). Além disso, o censo mais recente data de 2023.

Sobre a limitação técnica, destaco a inexperiência com o uso de _data lakes_, visto que estou começando nessa área agora e ainda não aprendi Pandas, por isso me senti mais segura usando o SQL. Além disso, o próprio Databricks na versão gratuita não ajuda muito, visto que a cada hora precisamos criar um novo _cluster_ e, no meu caso, ainda tive que criar as tabelas várias vezes a cada vez que perdia o _cluster_, pois tentei chamar as tabelas via DBFS com o Pandas, mas não deu certo. Contudo, foi interessante ver o resultado final.
